policy/resolver.go

```diff
@@ -1271,36 +1262,19 @@ func (s *LocalServices) jobsToQueries(ctx context.Context, policyMrn string, cac
 				}
 				arr.Items = append(arr.Items, rj.Uuid)
 
-				// process all the datapoints of this job
-				for dp := range executionQuery.Datapoints {
-					datapointID := executionQuery.Datapoints[dp]
-					datapointInfo, ok := collectorJob.Datapoints[datapointID]
-					if !ok {
-						return nil, nil, errors.New("failed to identity datapoint in collectorjob")
-					}
-
-					datapointInfo.Notify = append(datapointInfo.Notify, rj.Uuid)
-					rj.Datapoints[datapointID] = true
+				if err = connectDatapointsToReportingJob(executionQuery, rj, collectorJob.Datapoints); err != nil {
+					return nil, nil, err
 				}
 
 				continue
 			}
 
 			// (3) Data Queries handling
-			for _, parentID := range rj.Notify {
-				parentJob, ok := collectorJob.ReportingJobs[parentID]
-				if !ok {
-					return nil, nil, errors.New("failed to connect datapoint to reporting job")
-				}
-
-				if err = connectDatapointsToReportingJob(executionQuery, parentJob, collectorJob.Datapoints); err != nil {
-					return nil, nil, err
-				}
-
-				// we don't need this any longer, since every datapoint now reports into
-				// the parent job (i.e. reports into the policy instead of the data query)
-				delete(collectorJob.ReportingJobs, rj.Uuid)
-				delete(parentJob.Datapoints, rj.Uuid)
+			// We connect the datapoints to the data query reporting job. Previously we had connected the datapoints
```

Comment:
This is the biggest change. We don't remove the data query's RJ. Instead if we connect the datapoint to it.